src/Library/DeviceBase.cpp

```diff
@@ -90,14 +90,20 @@ namespace usbguard
      * Set connect type
      */
     int retry = 0;
-    std::string connect_type;
-    while (retry < 20) {
+    std::string connect_type = "";
+    /*
+     * Read port/connect_type only for non Host Controller devices
+     */
+    if (getPort().substr(0, 3) != "usb") {
+      while (retry < 10) {
```

Comment:
Based on your screeenshot with real life Pi durations, I suggest `retry < 30` (and `usleep(1000 * 1000);` below) to sleep up to 30 milliseconds total.

Reply:
Yes, I will update retry to 30 and usleep to 1000 (not 1000 * 1000) since usleep()  suspends execution for microsecond intervals (not nanoseconds).

Reply:
My bad, good point! :+1: